middleware/realip.go

```diff
@@ -40,21 +42,25 @@ func RealIP(h http.Handler) http.Handler {
 }
 
 func realIP(r *http.Request) string {
-	var ip string
+	if tcip := r.Header.Get(trueClientIP); isValidIP(tcip) {
+		return tcip
+	}
+
+	if xrip := r.Header.Get(xRealIP); isValidIP(xrip) {
+		return xrip
+	}
 
-	if tcip := r.Header.Get(trueClientIP); tcip != "" {
-		ip = tcip
-	} else if xrip := r.Header.Get(xRealIP); xrip != "" {
-		ip = xrip
-	} else if xff := r.Header.Get(xForwardedFor); xff != "" {
+	if xff := r.Header.Get(xForwardedFor); xff != "" {
 		i := strings.Index(xff, ",")
 		if i == -1 {
 			i = len(xff)
 		}
-		ip = xff[:i]
+		return xff[:i]
```

Comment:
looks like you're not checking if XFF is valid IP

Reply:
I greatly appreciate you sharing your observation. I will review the code and implement the necessary checks to ensure that the IP address present in the X-Forwarded-For (XFF) header is validated correctly. Thank you for bringing this to my attention!

Reply:
I greatly appreciate you sharing your observation. I will review the code and implement the necessary checks to ensure that the IP address present in the X-Forwarded-For (XFF) header is validated correctly. Thank you for bringing this to my attention!